internal/controlplane/handlers_user.go

```diff
@@ -483,3 +491,33 @@ func isUserSelfResolving(ctx context.Context, store db.Store, i db.GetInvitation
 
 	return nil
 }
+
+// generateRandomString is used to generate a random string of a given length
+func generateRandomString(length int) (string, error) {
+	bytes := make([]byte, length)
+	_, err := rand.Read(bytes)
+	if err != nil {
+		return "", err
+	}
+	return hex.EncodeToString(bytes)[:length], nil
+}
+
+// getUniqueProjectBaseName is used to generate a unique project name
+func getUniqueProjectBaseName(ctx context.Context, store db.Store, baseName string) (string, error) {
+	uniqueBaseName := baseName
+	for {
```

Comment:
I know it's unlikely we hit a really infinite loop here, but ` for {}` always makes me nervous. Do you think we could add some upper limit?

Reply:
I'll add a 10 times retry mechanism. It should be more than enough 👍